libshaderc/src/shaderc_cpp_test.cc

```diff
@@ -504,6 +505,106 @@ TEST_F(CppInterface, PreprocessingOnlyModeSecondOverridesDisassemblyMode) {
               HasSubstr("void main(){ }"));
 }
 
+// A mock class that simulate a client of the includer.
+class TestIncluder : public shaderc::CompileOptions::Includer {
+ public:
+  // Create a fake file object, with specified path and content.
+  struct FakeFile {
+    std::string path;
+    std::string content;
+  };
+  // Use hashmap as a fake file system to store fake files to be included.
+  using FakeFS = std::unordered_map<std::string, FakeFile>;
+
+  TestIncluder(FakeFS& fake_fs) : fake_fs_(fake_fs){};
+
+  // Get path and content from the fake file system.
+  shaderc_includer_response* GetIncluderResponse(
+      const char* filename) override {
+    FakeFile& file_to_include = fake_fs_[filename];
+    response_.path = file_to_include.path.c_str();
+    response_.path_length = file_to_include.path.size();
+    response_.content = file_to_include.content.c_str();
+    response_.content_length = file_to_include.content.size();
+    return &response_;
+  };
+
+  // Response data is owned as private property, no need to release explicitly.
+  void ReleaseIncluderResponse(shaderc_includer_response* data) override {
+    (void)data;
```

Comment:
Shouldn't have to do the (void)data thing.   To avoid a warning, delete the parameter name, i.e.
do:
```
ReleaseIncluderResponse(shaderc_includer_response*) override {}
```